### ISSUE TITLE
Specify package archive with `:ensure (t . pin)`

### ISSUE THREAD

**Issue by toyboot4e** [conao3/leaf.el#543]
## Description

Like [`:pin` in `use-package`](https://github.com/jwiegley/use-package/blob/a6e856418d2ebd053b34e0ab2fda328abeba731c/use-package-core.el#L1676), `:ensure (t . pin)` should pin a package to an archive. The syntax is ready, but the handler is not yet implemented.

## Issue leaf-block

Example:

```elisp
(leaf hoge
  :ensure (t . melpa-stable))
```

## Expected output from macroexpand-1 of the leaf-block

The macro expansion seems like already OK:

```elisp
(prog1 'hoge
  (leaf-handler-leaf-path hoge)
  (leaf-handler-leaf-protect hoge
      (leaf-handler-package hoge hoge melpa-stable) (leaf-handler-package hoge hoge nil)))
```

Just the implementations of [`leaf-handler-package`](https://github.com/conao3/leaf.el/blob/466e12a4161fdee53bca60a786a460b4680255f1/leaf.el#L1090) and [`leaf-options-ensure-default-pin`](https://github.com/conao3/leaf.el/blob/466e12a4161fdee53bca60a786a460b4680255f1/leaf.el#L432) are lacking.

## Related

- [`use-package-ensure-elpa`](https://github.com/jwiegley/use-package/blob/a6e856418d2ebd053b34e0ab2fda328abeba731c/use-package-ensure.el#L153) in `use-package` could be seen as a reference implementation.

**Comment by toyboot4e** [comment]
No, the macro expansion should be like this, [as in leaf-tests.el](https://github.com/conao3/leaf.el/blob/466e12a4161fdee53bca60a786a460b4680255f1/leaf-tests.el#L275)?

```elisp
;; leaf block
(leaf hoge
  :ensure (t . melpa-stable))
```

```diff
;; expanded
(prog1 'hoge
  (leaf-handler-leaf-path hoge)
  (leaf-handler-leaf-protect hoge
-      (leaf-handler-package hoge hoge melpa-stable) (leaf-handler-package hoge hoge nil)))
+      (leaf-handler-package hoge hoge melpa-stable)))
```

**Comment by conao3** [comment]
Maybe we should implement this missing feature on `leaf-handler-package`, so we have to add testcase on this.
https://github.com/conao3/leaf.el/blob/466e12a4161fdee53bca60a786a460b4680255f1/leaf-tests.el#L2405-L2431
